client/src/services/filter-helper.service.ts

```diff
@@ -0,0 +1,29 @@
+import { Filter } from '@/models/filter';
+
+export default class FilterHelperService {
+    static isValid(filter:Filter): boolean {
+
+        let year1, year2;
+        switch (filter.type) {
+            case 'date':
+                [year1, year2] = filter.value.split('-').map(t => parseInt(t, 10));
+                return FilterHelperService.isValidYear(year1) && FilterHelperService.isValidYear(year2) && year2 > year1;
+            case 'empty':
+                return false;
+            default: return !!filter.value;
+        }
+    }
+    static isValidYear(year: number) {
```

Comment:
The formatting seems a little bit off. Could you apply the prettier rules here? Also It would be nice to have jsdoc-style comments on each function to describe what its purpose is